include/superior_mysqlpp/extras/prepared_statement_utils.hpp

```diff
@@ -131,10 +93,72 @@ namespace SuperiorMySqlpp
              typename ConnType>
     void psReadQuery(const std::string &query, ConnType &&connection, Callable &&processingFunction)
     {
-        auto ps = detail::generatePreparedStatementImpl<storeResult, validateMode, warnMode, ignoreNullable>(connection, query, processingFunction);
+        auto ps = detail::generatePreparedStatement<storeResult, validateMode, warnMode, ignoreNullable>(connection, query, processingFunction, [](){});
         psReadQuery(ps, processingFunction);
     }
 
+    /**
+     * @brief Builds prepared statement from query string and constructs prepared stament with updatable arguments
+     * @param query Query to be executed (in most cases there will be selections)
+     * @param connection Connection handle into database
+     * @param paramsSetter function setting statement's parameters, returning bool if input data are available
+     * @param processingFunction function to be invoked on every row
+     *                           Its parameters must correspond with result columns (their types and count)
+     * @tparam storeResult Boolean indicating if results will be in `store` or `use` mode
+     * @tparam validateMode Indicates validate mode level
+     * @tparam warnMode Indicates warning mode level
+     * @tparam ignoreNullable Disables null type checking
+     */
+    template<bool storeResult=detail::PreparedStatementsDefault::getStoreResult(),
+             ValidateMetadataMode validateMode=detail::PreparedStatementsDefault::getValidateMode(),
+             ValidateMetadataMode warnMode=detail::PreparedStatementsDefault::getWarnMode(),
+             bool ignoreNullable=detail::PreparedStatementsDefault::getIgnoreNullable(),
+             typename ResultCallable,
+             typename ParamCallable,
+             typename ConnType>
```

Comment:
I don't see a point for templating here.

Reply:
Well, you can call it then like this:
```c++
psQuery("SELECT ...", gimmeConnection(), ....);
```
Or in another words, it gives you ability to use one-time-use connection.

Reply:
Fair enough.